pkg/debug/start_debug.go

```diff
@@ -0,0 +1,167 @@
+/*
+Copyright 2023 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package debug
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8s "k8s.io/client-go/kubernetes"
+)
+
+func StartDebug(clusterNamespace, deploymentName, flagValue string) {
+	// Get Kubernetes Client
+	_, clientset, _ := k8sutil.GetKubeClient()
+
+	deployment, err := verifyDeploymentExists(clientset, clusterNamespace, deploymentName)
+	if err != nil {
+		fmt.Printf("Missing mon or osd deployment name %s. %v\n", deploymentName, err)
+		os.Exit(1)
+	}
+
+	deploymentSpec := deployment.Spec
+
+	if flagValue != "" {
+		deploymentSpec.Template.Spec.Containers[0].Image = flagValue
+	}
+
+	labels := deployment.Labels
+	labels["ceph.rook.io/do-not-reconcile"] = "true"
+
+	deploymentSpec.Template.Spec.Containers[0].LivenessProbe = nil
+	deploymentSpec.Template.Spec.Containers[0].StartupProbe = nil
+
+	if flagValue != "" {
+		log.Printf("setting debug image to %s\n", flagValue)
+		deploymentSpec.Template.Spec.Containers[0].Image = flagValue
+	}
+
+	fmt.Println("setting debug command to main container")
+
+	deploymentSpec.Template.Spec.Containers[0].Command = []string{"sleep", "infinity"}
+	deploymentSpec.Template.Spec.Containers[0].Args = []string{}
+
+	if err := updateDeployment(clientset, clusterNamespace, deployment); err != nil {
+		fmt.Printf("Failed to update deployment %s. %v\n", deployment.Name, err)
+		os.Exit(1)
+	}
+
+	deployment_Pod_Name, err := podRunningWithLabel(clientset, clusterNamespace, deploymentSpec)
+	if deployment_Pod_Name == "" || err != nil {
+		os.Exit(1)
```

Comment:
can we print an error?

Reply:
done

Reply:
Now we can return the err instead of printing it
```suggestion
		return err
```